src/main/webapp/app/exam/participate/exam-navigation-bar/exam-navigation-bar.component.ts

```diff
@@ -27,9 +30,31 @@ export class ExamNavigationBarComponent implements OnInit {
 
     icon: IconProp;
 
-    constructor(private layoutService: LayoutService) {}
+    exerciseIdToNavigateTo: number;
+    subscriptionToExerciseNavigation: Subscription;
+
+    exerciseIdToUpdate: number;
+    exerciseProblemStatement: string;
+    subscriptionToLiveExamExerciseUpdates: Subscription;
+
+    constructor(
+        private layoutService: LayoutService,
+        private examExercisesNavigationService: ExamExercisesNavigationService,
+        private liveExamExerciseUpdateService: LiveExamExerciseUpdateService,
+    ) {}
 
     ngOnInit(): void {
+        this.subscriptionToExerciseNavigation = this.examExercisesNavigationService.currentExerciseIdToNavigateTo.subscribe((exerciseId) => {
+            this.exerciseIdToNavigateTo = exerciseId;
+            this.changeExerciseById(exerciseId);
```

Comment:
do I get this right: will one be directly be navigated to the updated exercise when there is an update in the statement of the exercise? I think this might lead to confusion during the exam.

Reply:
Yes, one will get navigated to the exercise, but only if/after the student clicks on the popUp notification.
This subscription should only trigger when the `navigateToTarget` method inside `notification-popup.component` is executed, i.e. once the pop-up is clicked.
![image](https://user-images.githubusercontent.com/65814168/124463202-d39a4a80-dd92-11eb-9e4a-fe8aa90a8a82.png)

Thus, I think there should not be the danger of a sudden, involuntary navigation for the students.

Reply:
ok, I see.